corehq/apps/celery/shared_task.py

```diff
@@ -28,6 +30,10 @@ def task(*args, **kwargs):
 
     kwargs.setdefault('serializer', 'json')
     kwargs.setdefault('queue', default_queue)
+    kwargs.setdefault('options', {})
+
+    if kwargs.get('base') == PeriodicTask:
+        kwargs['options']['queue'] = kwargs.get('queue')
```

Comment:
Why do periodic tasks use ` kwargs['options']['queue']` vs ` kwargs['queue']` like shared task?

Reply:
Ohh! Beat uses `kwargs['options']['queue']` in order to determine which queue would Periodic tasks should run on.
But for shared tasks `kwargs['queue']` would work as celery is directly running them.